src/functional/ffront/foast_passes/unroll_power_op.py

```diff
@@ -0,0 +1,55 @@
+from __future__ import annotations
+
+from typing import cast
+
+import eve
+from functional.common import GTSyntaxError
+from functional.ffront import common_types as ct, field_operator_ast as foast, type_info
+
+
+class UnrollPowerOp(eve.NodeTranslator):
+    @classmethod
+    def apply(cls, node: foast.LocatedNode):
+        return cls().visit(node)
+
+    def visit_BinOp(self, node: foast.BinOp) -> foast.BinOp:
+        if node.op == foast.BinaryOperator.POW:
+            if not type(node.right) is foast.Constant:
+                raise FieldOperatorPowerError.from_powOp_node(
+                    node,
+                    msg=f"'Power value '{type(node.right)}' not supported. Only integer values greater than zero allowed in the power operation",
+                )
+            if type_info.extract_dtype(cast(ct.ScalarType, node.right.type)).kind not in [
+                ct.ScalarKind.INT32,
+                ct.ScalarKind.INT64,
+            ]:
+                raise FieldOperatorPowerError.from_powOp_node(
+                    node, msg="Only integer values allowed in the power operation"
+                )
+            if int(node.right.value) == 0:
+                raise FieldOperatorPowerError.from_powOp_node(
+                    node,
+                    msg=f"'Power value '{type(node.right)}' not supported. Only integer values greater than zero allowed in the power operation",
```

Comment:
```suggestion
                    msg=f"'Only integer values greater than zero allowed in the power operation.",
```
Can be combined with the above error.